api/src/features/todo/use_cases/delete_todo_by_id.py

```diff
@@ -1,14 +1,14 @@
 from pydantic import BaseModel, Field
 
-from infrastructure.repositories.TodoRepository import TodoRepository
+from features.todo.interfaces.TodoRepositoryInterface import TodoRepositoryInterface
 
 
 class DeleteTodoByIdResponse(BaseModel):
     success: bool = Field(...)
 
 
-def delete_todo_use_case(id: str, todo_item_repository=TodoRepository()) -> DeleteTodoByIdResponse:
-    if not todo_item_repository.get(id):
+def delete_todo_use_case(id: str, todo_repository: TodoRepositoryInterface) -> DeleteTodoByIdResponse:
+    if not todo_repository.get(id):
```

Comment:
Not part of this PR. But IMO this should either return 404 or success=True

Reply:
The mongo database client will throw a NotFoundException, so this should already be handled

Reply:
If that is the case, the entire if statement is redundant

Reply:
Should be fixed, adding a issue for this